persistence-api/src/main/java/io/stargate/db/datastore/PersistenceBackedResultSet.java

```diff
@@ -104,7 +106,15 @@ static ResultSet create(
 
   private void processNewPage(Result.Rows page) {
     for (List<ByteBuffer> rowValues : page.rows) {
-      fetchedRows.addLast(new ArrayListBackedRow(columns, rowValues, driverProtocolVersion));
+      ArrayListBackedRow row = new ArrayListBackedRow(columns, rowValues, driverProtocolVersion);
+
+      if (authzFilter != null) {
+        if (authzFilter.test(row)) {
+          fetchedRows.addLast(row);
+        }
+      } else {
+        fetchedRows.addLast(row);
+      }
```

Comment:
nit: how about:
```
if (authzFilter == null || authzFilter.test(row)) {
  fetchedRows.addLast(row);
}
```
As for me, it is easier to read :)

Reply:
Fixed.